src/ads/MobileAd.ts

```diff
@@ -208,12 +220,22 @@ export abstract class MobileAd implements MobileAdInterface {
   public abstract addAdEventListener<T extends never>(type: T, listener: AdEventListener<T>): void;
 
   public removeAllListeners() {
+    this._checkDestroyed();
     this._adEventsListeners.clear();
     this._adEventListenersMap.forEach((_, type, map) => {
       map.set(type, new Map());
     });
   }
 
+  public destroy() {
+    this._checkDestroyed();
+    const destroy = this._googleMobileAds.native[`${this._camelCaseType}Destroy`];
+    destroy(this._requestId);
```

Comment:
This seems like it would work but it was confusing to the reader (me) to have the method called destroy, and the returned function also called destroy in local scope. At first I thought "Is this recursive?" 

suggest renaming to nativeDestroyCallback or something for easy reading comprehension

Reply:
It makes sense.